adam/experiment/live_demo.py

```diff
@@ -0,0 +1,319 @@
+import argparse
+import base64
+import json
+import pickle
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+
+import cv2
+import numpy as np
+import requests
+import yaml
+from vistautils.parameters import Parameters
+
+from adam.curriculum.curriculum_from_files import phase3_process_scene_dir
+from adam.experiment.observer import YAMLLogger
+from adam.learner.integrated_learner import SimulatedIntegratedTemplateLearner
+from adam.paths import (
+    TESTING_CURRICULUM_DIR,
+    LEARNERS_DIR,
+    PREPROCESSING_DIR,
+    TRAINING_CURRICULUM_DIR,
+)
+
+MENU_STR = "ADAM Demo UI Valid Commands: (d)ecode, (e)xit, (r)eset curriculum: "
+DECODE_STR = "Filepath to the image to decode: "
```

Comment:
For clarity I'd call these `MENU_PROMPT` and `DECODE_PROMPT` instead. `STR` is redundant since we're assigning literal values and `PROMPT` is more specific about how they're used.